x-pack/plugins/alerting/server/rules_client/lib/siem_legacy_actions/format_legacy_actions.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { chunk } from 'lodash';
+import type { SavedObjectsFindOptionsReference, Logger } from '@kbn/core/server';
+import pMap from 'p-map';
+import { RuleAction, Rule } from '../../../types';
+import type { RuleExecutorServices } from '../../..';
+import { injectReferencesIntoActions } from '../../common';
+import { transformToNotifyWhen } from './transform_to_notify_when';
+import { transformFromLegacyActions } from './transform_legacy_actions';
+import { LegacyIRuleActionsAttributes, legacyRuleActionsSavedObjectType } from './types';
+
+/**
+ * @deprecated Once we are confident all rules relying on side-car actions SO's have been migrated to SO references we should remove this function
+ */
+interface LegacyGetBulkRuleActionsSavedObject {
+  alertIds: string[];
+  savedObjectsClient: RuleExecutorServices['savedObjectsClient'];
+  logger: Logger;
+}
+
+/**
+ * @deprecated Once we are confident all rules relying on side-car actions SO's have been migrated to SO references we should remove this function
+ */
+export interface LegacyActionsObj {
+  ruleThrottle: string | null;
+  legacyRuleActions: RuleAction[];
+}
+
+/**
+ * @deprecated Once we are confident all rules relying on side-car actions SO's have been migrated to SO references we should remove this function
+ */
+export const legacyGetBulkRuleActionsSavedObject = async ({
```

Comment:
Note: one thing I wish we'd done the first time around writing the migration code is to have better documented _everything_ . So even though it may seem obvious right now what each of these do as we're working on it, you (general "you") will soon forget. 

Maybe adding a description of what each function does, however obvious it may seem, will help our future selves. Some additional documenting that may help is a note as to why we only search by `siem-detection-engine-rule-actions` and not also `siem.notifications`. I outlined it in this PR - pre 7.16, even rules created with no actions or actions for every rule run, generate a `siem-detection-engine-rule-actions` but not a `siem.notifications` SO, so we cannot rely on `siem.notifications` alone.

Reply:
thanks for the suggestion.
I put some additional comments to functions and reference to that PR as well